File: backend service/src/main/java/mutrivia/demo/backend/model/Question.java
package mutrivia.demo.backend.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import lombok.*;
import org.hibernate.annotations.GenericGenerator;

import java.util.List;
@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Question {
    @Id
    @GeneratedValue(generator="system-uuid")
    @GenericGenerator(name="system-uuid", strategy = "uuid")
    private String questionId;
    @JsonProperty("questionStatement")
    private String questionStatement;
    @JsonProperty("options")
    private List<String> options;
    @JsonProperty("correctChoiceIndex")
    private int correctChoiceIndex;
    @JsonProperty("textDataId")
    private String textDataId;
}

File: frontend/mutrivia-flutter/lib/models/question.dart
class Question {
  final String questionStatement;
  final List<String> options;
  final int correctChoiceIndex;
  final String userId;

  const Question({
    required this.questionStatement,
    required this.options,
    required this.correctChoiceIndex,
    required this.userId,
  });

  factory Question.fromJson(Map<String, dynamic> json) {
    return Question(
      questionStatement: json['questionStatement'] as String,
      options: json['options'] as List<String>,
      correctChoiceIndex: json['correctChoiceIndex'] as int,
      userId: json['userId'] as String,
    );
  }
}